xdem/spstats.py

```diff
@@ -9,19 +9,92 @@
 import random
 import warnings
 from functools import partial
-from typing import Callable, Union
+from typing import Callable, Union, Optional
 
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 from scipy import integrate
 from scipy.optimize import curve_fit
+from skimage.draw import disk
+from scipy.interpolate import RegularGridInterpolator, LinearNDInterpolator, griddata
+from xdem.spatial_tools import nmad
 
 with warnings.catch_warnings():
     warnings.filterwarnings("ignore", category=DeprecationWarning)
     import skgstat as skg
     from skgstat import models
 
+def robust_interp_nd_binning(df: pd.DataFrame, var_names: list[str], statistic : Union[str, Callable, None] = nmad, min_count: Optional[int] = 100) -> Callable:
+    """
+    Estimate an interpolated function for an N-dimensional binning.
+    Relies on scipy.ndimage.map_coordinates: see for more information on input arguments.
+
+    :param df: dataframe of binned values according to explanatory variables
+    :param var_names: list of variable names to use
+    :param statistic: statistic to use
```

Comment:
Wait, why could `statistic` be a callable if it's never called? The only thing that is used is the `__name__` property, so wouldn't it be easier to just allow `str | None`?

Reply:
Oh, and what happens if it's `None`?

Reply:
Here again it's for consistency with the input `statistic` of `nd_binning`.
The name might not be obvious to user, more complex to input, for instance: `count` name is `count` but `np.nanmedian` is `nanmedian`.

Reply:
Could this be elaborated upon in the docstring?